src/Distribution/Server/Packages/Unpack.hs

```diff
@@ -468,6 +480,10 @@ explainTarError (PermissionsError entryname mode) =
   where
     showMode :: Tar.Permissions -> String
     showMode m = printf "%.3o" (fromIntegral m :: Int)
+explainTarError (LongNamesError err) =
+    "There is an error in the format of entries with long names in the tar file: " ++ show err
+ ++ ". Check that it is a valid tar file (e.g. 'tar -xtf thefile.tar'). "
+ ++ "You may need to re-create the package tarball and try again."
```

Comment:
Can we get a test case that triggers this error?  

Could be not easy to engineer, but it is good to have evidence for every error that it is possible to trigger.

Reply:
Are there tests for pre-existing errors somewhere? I can add a new one there.

This is a rare condition though: one has to have an intent to craft a malicious TAR, you cannot obtain one by mistake.

Reply:
> Are there tests for pre-existing errors somewhere? I can add a new one there.

Yeah, that is the usual problem: There aren't enough tests to start with, so there isn't a good pretext to add some for new or changed features.  One can't require adding a test infrastructure for a maintenance PR like this.